### DISCUSSION TITLE
Is it possible to use asynchronous functions in/as http handlers?

### DISCUSSION THREAD

**Discussion by AlexTolmachev** [uNetworking/uWebSockets.js#50] [Q&A] [answered]
This code causes server to terminate:
```javascript
app.get('/*', async (res) => {
  res.end(await someFunc());
});
```

**Comment by ghost** [comment, marked as answer]
You have two problems here - you can't return from a handler without responding OR attaching an abort handler, just like the message you got says. This message probably needs to be simplified, but it should terminate.

Second, it is now possible to use async/await as per last commit, I tested it and it works.

**Comment by ghost** [comment]
This could use an example - AsyncFunction.js or something

**Comment by ghost** [comment]
I have added an example for this.

**Comment by AlexTolmachev** [comment]
Thanks a lot! It’s very good to see, that library is under active development 🥳

**Comment by ghost** [comment]
I sure hope so. First "public" release was 5 days ago 😉
